src/metabase/api/action.clj

```diff
@@ -139,4 +143,28 @@
         (db/update! existing-model id type-row))))
   (first (action/actions-with-implicit-params nil :id id)))
 
+(api/defendpoint POST "/:id/public_link"
+  "Generate publicly-accessible links for this Action. Returns UUID to be used in public links. (If this
+  Action has already been shared, it will return the existing public link rather than creating a new one.) Public
+  sharing must be enabled."
+  [id]
+  (api/check-superuser)
+  (validation/check-has-application-permission :setting)
+  (validation/check-public-sharing-enabled)
+  (api/read-check Action id)
+  {:uuid (or (db/select-one-field :public_uuid Action :id id)
+             (u/prog1 (str (UUID/randomUUID))
+                      (db/update! Action id
+                                  :public_uuid <>
+                                  :made_public_by_id api/*current-user-id*)))})
+
+(api/defendpoint DELETE "/:id/public_link"
+  "Delete the publicly-accessible link to this Dashboard."
+  [id]
```

Comment:
should we require admin for this?

Reply:
See this comment
https://github.com/metabase/metabase/blob/f02a110b5588c2218c81b86cf44ea7ae00cccd25/src/metabase/api/action.clj#L166